ingester2/src/persist/on_demand.rs

```diff
@@ -0,0 +1,44 @@
+use crate::{
+    partition_iter::PartitionIter,
+    persist::{drain_buffer::persist_partitions, queue::PersistQueue},
+};
+use generated_types::influxdata::iox::ingester::v1::{
+    self as proto, persist_service_server::PersistService,
+};
+use tonic::{Request, Response};
+
+#[derive(Debug)]
+pub(crate) struct PersistNow<T, P> {
+    buffer: T,
+    persist_handle: P,
+}
+
+impl<T, P> PersistNow<T, P>
+where
+T: PartitionIter + Sync + 'static,
+P: PersistQueue + Clone + Sync + 'static,
+ {
+    pub(crate) fn new(buffer: T, persist_handle: P) -> Self {
+        Self {
+            buffer,
+            persist_handle,
+        }
+    }
+}
+
+#[tonic::async_trait]
+impl<T, P> PersistService for PersistNow<T, P>
+where
+T: PartitionIter + Sync + 'static,
+P: PersistQueue + Clone + Sync + 'static,
+ {
+    /// Handle the RPC request to persist immediately.
+    async fn persist(
+        &self,
+        _request: Request<proto::PersistRequest>,
+    ) -> Result<Response<proto::PersistResponse>, tonic::Status> {
+        persist_partitions(self.buffer.partition_iter(), self.persist_handle.clone()).await;
```

Comment:
Very minor nit: after a80e66336bd55851ad81b01bbef4ca09fcb5e645 I think this can be:

```suggestion
        persist_partitions(self.buffer.partition_iter(), &self.persist_handle).await;
```